splice_machine/src/test/java/com/splicemachine/derby/test/framework/SpliceWatcher.java

```diff
@@ -51,6 +54,8 @@
 public class SpliceWatcher extends TestWatcher implements AutoCloseable {
 
     private static final Logger LOG = Logger.getLogger(SpliceWatcher.class);
+    private static ExecutorService executor = Executors.newCachedThreadPool(
+            new ThreadFactoryBuilder().setDaemon(true).setNameFormat("connection-close-%d").build());
```

Comment:
wouldn't this cause problems when running tests in parallel?

Reply:
I don't think so, each task would get it's own thread and we are not limiting the amount of threads, so there should be no conflict.

Reply:
right, ok!